pywbem/_server.py

```diff
@@ -885,6 +943,146 @@ def get_central_instances(self, profile_path, central_class=None,
+        required because there is an issue between implementations in the usage
+        of the dependent and antecedent reference properties. Typically SNIA
+        profiles defined them with:
+
+            Antecedent = referenced profile = component profile
+            Dependent = referencing profile = autonomous profile
+
+        and many SMI implementations defined them with:
+
+            Antecedent = autonomous profile
+            Dependent = component (= sub) profile
+
+        This method processes the CIM_RegisteredProfiles associatorNames
+        operation with ResultRole set to `Dependent` and then to `Antecedent` to
+        find any profile returns associations for one of the ResultRole
+        settings but not the other. If such an association can be found, that
+        determines the implementation of the antecedent/dependent direction by
+        matching the returns against possible_autonomous_profiles and
+        possible_component_profiles.  Either of the lists or both may be
+        provided. At least one of the lists must exist.
+
+        If the test determines the direction, it returns a string defining the
+        direction (`dmtf` or `snia`). If no direction can be found the test
+        returns `None`.  If the test has inconsistent results it raises an
+        exception.
+
+          Parameters:
+
+            possible_autonomous_profiles (list of :class:`~pywbem.CIMInstanceName`))
+                List of registered profile CIMInstanceName objects that are
+                autonomous profiles. This list is optional but either this list
+                or  the possible_component_profiles list must exist.
+
+            possible_component_profiles (list of :class:`~pywbem.CIMInstanceName`))
+                List of registered profile CIMInstanceName objects that are
+                autonomous profiles. This list is optional but either this list
+                or  the possible_autonomous_profiles list must exist.
+
+          Returns: (:term:`string` or None)
+            Where the string can be either 'dmtf' or 'snia' if the test
+            determines that the antecedent/dependent definitions match the
+            possible profiles provided.
+
+            None is returned if the test can not find any profiles in the server
+            that match the profile names in either of the lists input.
+
+          Raises:
+            ValueError: If the test finds inconsistent results. This can include
+            the possible_autonomous_profiles list generating one definition and
+            the possible_component_profiles the opposite definition, or one of
+            the lists concluding that the possible direction could be both
+            'dmtf' and 'snia'because of inconsistencies in the instances of
+            CIM_ReferencedProfile
+        """  # noqa: E501
+        # pylint: enable=line-too-long
+
+        def _determine_type(profilepaths, v0_dict, v1_dict, autonomous):
```

Comment:
Local function definitions should only be used if there is a reason, i.e. if they use local variables of the outer function. That does not seem to be the case here. So I suggest to make this a classmethod in the class.

Reply:
made it staticmethod since it makes no use of anythin in the class.

Reply:
sorry, yes, that‘s what I meant